content/gocourse-2019-05-melbourne.slide

```diff
@@ -1142,6 +1142,35 @@ Rob Pike's Talk ([[https://blog.golang.org/toward-go2][Towards Go 2]])
 Dave Cheney Blog ([[https://dave.cheney.net/paste/the-past-present-and-future-of-go2.pdf][Past Present and Future of Go]])
 
 
+* Go for Java devs
+
+- Functions are lambdas
+- Typed constants are enums
+- exported = public
+- unexported = protected, not private
+- struct != class
+An empty struct is pointless
```

Comment:
Why?

Reply:
Original thinking was aliasing `struct{}` doesn't do much.
```
type Foo struct{}
```
I'm not sure why you would do this
However, `struct{}` is used a lot (eg: `chan struct{}` for signalling between goroutines), so this point is misleading.
Open to being proven wrong on aliasing `struct{}` as well.

Reply:
How about to implement an interface where "data" is needed?